lib/Parse/ParseDecl.cpp

```diff
@@ -8980,6 +8980,18 @@ Parser::parseDeclEnumCase(ParseDeclOptions Flags,
       if (ArgParams.isNull() || ArgParams.hasCodeCompletion())
         return ParserStatus(ArgParams);
     }
+    
+    // See if there are generic params.
+    auto genericParams = maybeParseGenericParams()
+      .getPtrOrNull();
+    if (genericParams) {
+      auto param = genericParams->getParams().front()->getStartLoc();
```

Comment:
Just because we parsed a generic parameter clause doesn’t mean it’s non-empty. Please check for that

Reply:
Added validation for that check.

Reply:
With this change, could `case bar<>(param: Int)` pass through without any diagnostics? I think this should trap as well.

Reply:
`case bar<>(param: Int)` presents the message: `error: expected an identifier to name generic parameter`

Reply:
This diagnostic may be a little misleading as solving that error by adding a generic parameter name results in another error saying that generic parameters are not allowed. 

I think we could do better by emitting the same ` generic parameter not allowed` in such case as well. Any `<`(and or `>`) tokens preceding the case identifier would fall under this.